File: Java/ATM.java
import java.util.Scanner;

public class ATM {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        String nama;
        boolean validNama = false;

        // 1–3. Meminta nama sampai valid (TRUE)
        do {
            System.out.print("Masukkan Nama: ");
            nama = input.nextLine();

            System.out.print("Apakah nama sudah benar? (TRUE/FALSE): ");
            String konfirmasi = input.nextLine().toUpperCase();

            if (konfirmasi.equals("TRUE")) {
                validNama = true;
            } else {
                System.out.println("Nama tidak valid! Silakan masukkan ulang.\n");
            }

        } while (!validNama);

        // 4–5. Input NIM dan konversi menjadi saldo
        long nim;
        System.out.print("Masukkan NIM (angka saja): ");
        nim = input.nextLong();

        long saldo = nim; // misal NIM: 2310355 → Saldo: Rp 2.310.355

        // 6. Tampilkan Nama dan Saldo awal
        System.out.println("\n============================");
        System.out.println("Nama  : " + nama);
        System.out.println("Saldo : Rp " + saldo);
        System.out.println("============================\n");

        int pilihan;

        // 7. Menu ATM menggunakan Switch–Case
        do {
            System.out.println("===== MENU ATM =====");
            System.out.println("1. Cek Saldo");
            System.out.println("2. Tarik Tunai");
            System.out.println("3. Setor Tunai");
            System.out.println("4. Transfer");
            System.out.println("5. Keluar");
            System.out.print("Pilih Menu: ");
            pilihan = input.nextInt();

            switch (pilihan) {
                case 1:
                    System.out.println("Saldo Anda: Rp " + saldo);
                    break;

                case 2:
                    System.out.print("Masukkan jumlah tarik tunai: Rp ");
                    long tarik = input.nextLong();
                    if (tarik > saldo) {
                        System.out.println("Saldo tidak cukup!");
                    } else {
                        saldo -= tarik;
                        System.out.println("Tarik tunai berhasil! Saldo sekarang: Rp " + saldo);
                    }
                    break;

                case 3:
                    System.out.print("Masukkan jumlah setor tunai: Rp ");
                    long setor = input.nextLong();
                    saldo += setor;
                    System.out.println("Setor tunai berhasil! Saldo sekarang: Rp " + saldo);
                    break;

                case 4:
                    System.out.print("Masukkan jumlah transfer: Rp ");
                    long transfer = input.nextLong();
                    if (transfer > saldo) {
                        System.out.println("Saldo tidak cukup!");
                    } else {
                        saldo -= transfer;
                        System.out.println("Transfer berhasil! Saldo sekarang: Rp " + saldo);
                    }
                    break;

                case 5:
                    System.out.println("Terima kasih telah menggunakan ATM!");
                    break;

                default:
                    System.out.println("Menu tidak tersedia!");
            }

            System.out.println();

        } while (pilihan != 5);

        input.close();
    }
}

File: c++/ATM.cpp
#include <iostream>
#include <iostream>
#include <string>
using namespace std;

int main() {
    string nama, konfirmasi;
    bool validNama = false;

    // 1–3. Meminta nama sampai valid (TRUE)
    do {
        cout << "Masukkan Nama: ";
        getline(cin, nama);

        cout << "Apakah nama sudah benar? (TRUE/FALSE): ";
        getline(cin, konfirmasi);

        if (konfirmasi == "TRUE" || konfirmasi == "true") {
            validNama = true;
        } else {
            cout << "Nama tidak valid! Silahkan masukkan kembali.\n\n";
        }
    } while (!validNama);

    // 4–5. Input NIM → Saldo
    long long nim;
    cout << "Masukkan NIM (angka saja): ";
    cin >> nim;

    long long saldo = nim; // contoh: 2510381 → Rp 2.510.381

    // 6. Tampilkan Data Awal
    cout << "\n============================\n";
    cout << "Nama  : " << nama << endl;
    cout << "Saldo : Rp " << saldo << endl;
    cout << "============================\n\n";

    int pilihan;

    // 7. Menu ATM menggunakan switch–case
    do {
        cout << "===== MENU ATM =====\n";
        cout << "1. Cek Saldo\n";
        cout << "2. Tarik Tunai\n";
        cout << "3. Setor Tunai\n";
        cout << "4. Transfer\n";
        cout << "5. Keluar\n";
        cout << "Pilih Menu: ";
        cin >> pilihan;

        switch (pilihan) {
            case 1:
                cout << "Saldo Anda: Rp " << saldo << endl;
                break;

            case 2: {
                long long tarik;
                cout << "Masukkan jumlah tarik tunai: Rp ";
                cin >> tarik;

                if (tarik > saldo) {
                    cout << "Saldo tidak cukup!\n";
                } else {
                    saldo -= tarik;
                    cout << "Tarik tunai berhasil! Saldo sekarang: Rp " << saldo << endl;
                }
                break;
            }

            case 3: {
                long long setor;
                cout << "Masukkan jumlah setor tunai: Rp ";
                cin >> setor;

                saldo += setor;
                cout << "Setor tunai berhasil! Saldo sekarang: Rp " << saldo << endl;
                break;
            }

            case 4: {
                long long transfer;
                cout << "Masukkan jumlah transfer: Rp ";
                cin >> transfer;

                if (transfer > saldo) {
                    cout << "Saldo tidak cukup!\n";
                } else {
                    saldo -= transfer;
                    cout << "Transfer berhasil! Saldo sekarang: Rp " << saldo << endl;
                }
                break;
            }

            case 5:
                cout << "Terima kasih telah menggunakan ATM!\n";
                break;

            default:
                cout << "Menu tidak tersedia!\n";
        }

        cout << endl;

    } while (pilihan != 5);

    return 0;
}
